src/bumerang/borrowhandler.py

```diff
@@ -25,10 +27,13 @@ def get(self, id=None):
 
     def post(self):
         """Handle creating new requests and updating requests."""
-        request = self._create_borrow_node()
         try:
+            request = self._create_borrow_node()
             br_id = self.borrow_repo.insert_one(request)
             self.write({'id': br_id})
+        except ValueError as e:
```

Comment:
For the future, we should have `_create_borrow_node()` catch a value error and throw an error that inherits for bumerang error, but this is fine for now